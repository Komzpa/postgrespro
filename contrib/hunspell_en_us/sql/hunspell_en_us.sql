CREATE EXTENSION hunspell_en_us;

SELECT ts_lexize('english_hunspell', 'stories');
SELECT ts_lexize('english_hunspell', 'traveled');
SELECT ts_lexize('english_hunspell', 'eaten');
SELECT ts_lexize('english_hunspell', 'I''m');
SELECT ts_lexize('english_hunspell', 'Saturdays');
SELECT ts_lexize('english_hunspell', 'healthcare');
SELECT ts_lexize('english_hunspell', 'generally');
SELECT ts_lexize('english_hunspell', 'integrating');
SELECT ts_lexize('english_hunspell', 'lankiness''s');
SELECT ts_lexize('english_hunspell', 'rewritten');