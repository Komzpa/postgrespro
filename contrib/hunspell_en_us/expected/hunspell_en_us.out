CREATE EXTENSION hunspell_en_us;
CREATE TABLE table1(name varchar);
INSERT INTO table1 VALUES ('leaves'), ('leaved'), ('leaving'),
						('inability'), ('abilities'), ('disability'), ('ability');
SELECT d.* FROM table1 AS t, LATERAL ts_debug('english_hunspell', t.name) AS d;
   alias   |   description   |   token    |          dictionaries           |    dictionary    |  lexemes  
-----------+-----------------+------------+---------------------------------+------------------+-----------
 asciiword | Word, all ASCII | leaves     | {english_hunspell,english_stem} | english_hunspell | {leave}
 asciiword | Word, all ASCII | leaved     | {english_hunspell,english_stem} | english_hunspell | {leave}
 asciiword | Word, all ASCII | leaving    | {english_hunspell,english_stem} | english_hunspell | {leave}
 asciiword | Word, all ASCII | inability  | {english_hunspell,english_stem} | english_hunspell | {ability}
 asciiword | Word, all ASCII | abilities  | {english_hunspell,english_stem} | english_hunspell | {ability}
 asciiword | Word, all ASCII | disability | {english_hunspell,english_stem} | english_hunspell | {ability}
 asciiword | Word, all ASCII | ability    | {english_hunspell,english_stem} | english_hunspell | {ability}
(7 rows)

CREATE INDEX name_idx ON table1 USING GIN (to_tsvector('english_hunspell', "name"));
SELECT * FROM table1 WHERE to_tsvector('english_hunspell', name)
	@@ to_tsquery('english_hunspell', 'leaving');
  name   
---------
 leaves
 leaved
 leaving
(3 rows)

SELECT * FROM table1 WHERE to_tsvector('english_hunspell', name)
	@@ to_tsquery('english_hunspell', 'abilities');
    name    
------------
 inability
 abilities
 disability
 ability
(4 rows)

SELECT * FROM table1 WHERE to_tsvector('english_hunspell', name)
	@@ to_tsquery('english_hunspell', 'ability');
    name    
------------
 inability
 abilities
 disability
 ability
(4 rows)

DROP INDEX name_idx;
CREATE INDEX name_idx ON table1 USING GIST (to_tsvector('english_hunspell', "name"));
SELECT * FROM table1 WHERE to_tsvector('english_hunspell', name)
	@@ to_tsquery('english_hunspell', 'leaving');
  name   
---------
 leaves
 leaved
 leaving
(3 rows)

SELECT * FROM table1 WHERE to_tsvector('english_hunspell', name)
	@@ to_tsquery('english_hunspell', 'abilities');
    name    
------------
 inability
 abilities
 disability
 ability
(4 rows)

SELECT * FROM table1 WHERE to_tsvector('english_hunspell', name)
	@@ to_tsquery('english_hunspell', 'ability');
    name    
------------
 inability
 abilities
 disability
 ability
(4 rows)

